test/spi/spi_gaskets_test.dart

```diff
@@ -0,0 +1,638 @@
+
+  SpiPairTest(this.stimulus, super.name) : super() {
+    intf = SpiInterface(dataLength: 8);
+
+    monitor = SpiMonitor(intf: intf, parent: this);
+
+    Directory(outFolder).createSync(recursive: true);
+
+    final tracker =
+        SpiTracker(intf: intf, dumpTable: true, outputFolder: outFolder);
+
+    clk = SimpleClockGenerator(10).clk;
+
+    // initialize main
+    resetMain = Logic();
+    busInMain = Logic(width: 8);
+    starts = Logic();
+
+    main = SpiMain(intf,
+        busIn: busInMain, clk: clk, reset: resetMain, start: starts);
+
+    //init sub
+    resetSub = Logic();
+    busInSub = Logic(width: 8);
+    sub = SpiSub(intf: intf, busIn: busInSub, reset: resetSub);
+
+    // sub.busOut.changed.listen((_) {
+    //   logger.info('BusOut changed: ${sub.busOut.value}');
+    // });
+
+    intf.miso.changed.listen((_) {
+      logger.info('Miso changed: ${intf.miso.value}');
+    });
+
+    Simulator.registerEndOfSimulationAction(() async {
+      await tracker.terminate();
+    });
+
+    monitor.stream.listen(tracker.record);
+  }
+
+  @override
+  Future<void> run(Phase phase) async {
+    unawaited(super.run(phase));
+
+    final obj = phase.raiseObjection('SpiPairTestObj');
+
+    // Initialize all inputs to initial state.
+    // Just for waveform clarity.
+    await clk.waitCycles(1);
+
+    busInMain.inject(00);
+    busInSub.inject(00);
+
+    starts.inject(false);
+    resetMain.inject(false);
+    resetSub.inject(false);
+
+    await clk.waitCycles(1);
+    resetMain.inject(true);
+    resetSub.inject(true);
+
+    await clk.waitCycles(1);
+    resetMain.inject(false);
+    resetSub.inject(false);
+
+    await clk.waitCycles(1);
+    await stimulus(this);
+
+    obj.drop();
+  }
+}
+
+void main() {
+  tearDown(() async {
+    await Simulator.reset();
+  });
+
+  group('main gasket tests', () {
+    Future<void> runMainTest(SpiMainTest spiMainTest,
+        {bool dumpWaves = true}) async {
+      Simulator.setMaxSimTime(3000);
+
+      if (dumpWaves) {
+        await spiMainTest.main.build();
+        WaveDumper(spiMainTest.main,
+            outputPath: '${spiMainTest.outFolder}/waves.vcd');
+      }
+      await spiMainTest.start();
+    }
+
+    Future<void> sendMainData(SpiMainTest test, int data) async {
+      test.busInMain.inject(LogicValue.ofInt(data, test.intf.dataLength));
+      test.reset.inject(true);
+      await test.clk.nextPosedge;
+      test.reset.inject(false);
+      test.starts.inject(true);
+      await test.clk.waitCycles(1);
+      test.starts.inject(false);
+      await test.clk.waitCycles(7);
+    }
+
+    Future<void> sendSubPacket(SpiMainTest test, LogicValue data) async {
+      test.sub.sequencer.add(SpiPacket(data: data.reversed));
+      // TODO: fix reversed on subdriver
+      await sendMainData(test, 0x00);
+    }
+
+    test('simple transfers no gap', () async {
+      await runMainTest(SpiMainTest((test) async {
+        await sendSubPacket(test, LogicValue.ofInt(0x72, 8));
```

Comment:
we should add a check that there's no gap.  perhaps using timestamps from the monitor to make sure it's exactly 8 cycles apart in time?